user/overriding-packages.rst

```diff
@@ -0,0 +1,283 @@
+
+Install headers to a unique include directory
+*********************************************
+
+Install your package's headers to a unique folder rather than a shared folder.
+Say you're the author of a package ``foo``, and it has a header meant to be included like ``#include <foo/foo.hpp>``.
+Instead of installing the header to ``<prefix>/include/foo/foo.hpp``, install it to ``<prefix>/include/foo/foo/foo.hpp``.
+
+Here's an example.
+If your package ``foo`` has the directory structure ``include/foo/foo.hpp``, then in CMake it can install its headers to a unique directory with this:
+
+.. code-block:: CMake
+
+  install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
+
+All exported targets in your project need to export the unique include directory too.
+
+.. code-block:: CMake
+
+    target_include_directories(some_library_in_foo INTERFACE
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+      "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+
+Dynamically link to libraries outside your package
+**************************************************
+
+If your package ``foo`` statically links to ``libbar.lib`` from package ``bar``, then users can't override ``bar`` without also overriding yours.
+Prefer dynamic linking to ``libbar.so`` instead.
+
+Handling Python entry point specifications
+******************************************
+
+If your package loads Python entry points and it encounters two specifications with the same name, then it should use the last specification returned by `entry_points() <https://docs.python.org/3/library/importlib.metadata.html#entry-points>`_.
+It should also ignore entry points that can't be loaded.
+
+Here's how to do it:
+
+.. code-block:: Python
+
+    from importlib.metadata import entry_points
+
+    # Deduplicate entry point specifications before loading
+    deduplicated_entry_points = {}
+    # When faced with duplicates, this loop keeps the last entry point found
+    for ep in entry_points()['your_group_name']:
+        deduplicated_entry_points[ep.name] = ep
+
+    for ep in deduplicated_entry_points:
+        try:
+            inst = ep.load()
+        except ImportError:
+            # Ignore entry point specifications that can't be loaded
+            pass
+
+All Known issues
+----------------
+
+Include Directory Search Order Problem
+**************************************
+When overriding a package, it's possible for packages to find its headers from the underlay instead of the overlay.
+This may cause a failure to build or undefined behavior at runtime depending on the differences between those headers.
+
+Consider an overlay containing package ``foo`` and ``bar``, and an underlay containing ``bar`` and ``baz``.
+``foo`` depends on ``bar`` and ``baz``.
+Say the underlay is a merged workspace, and both the overriden ``bar`` and ``baz`` install their headers to a directory called ``include/``.
```

Comment:
Explain what a merged workspace is or link to an explanation.

Reply:
Added a page explaining merged workspaces to this PR, and a link to it in 2fe99b7ee29574a6dd693385c94777c532082dab